segmentstore/server/src/test/java/io/pravega/segmentstore/server/SegmentStoreMetricsTests.java

```diff
@@ -196,6 +198,23 @@ public void testCacheManagerMetrics() {
         assertNull(MetricRegistryUtils.getTimer(MetricsNames.CACHE_MANAGER_ITERATION_DURATION));
     }
 
+    @Test
+    public void testTreadPoolMetrics() throws InterruptedException {
+        @Cleanup("shutdown")
+        ScheduledExecutorService coreExecutor = ExecutorServiceHelpers.newScheduledThreadPool(30, "core", Thread.NORM_PRIORITY);
+        @Cleanup("shutdown")
+        ScheduledExecutorService storageExecutor = ExecutorServiceHelpers.newScheduledThreadPool(30, "storage-io", Thread.NORM_PRIORITY);
+
+        @Cleanup
+        SegmentStoreMetrics.ThreadPool pool = new SegmentStoreMetrics.ThreadPool(coreExecutor, storageExecutor);
+        Thread.sleep(2000);
```

Comment:
I think that sleeping in unit tests is not a good practice. I understand you do this because the reporter within the class you're testing executes once every 1 seconds.

Have you considered exposing the inner method that is invoked in that periodic runner (VisibleForTesting/PackagePrivate) and executing it manually in this test? That way you won't have to add this sleep here.

Reply:
Wondering if just using `assertEventuallyEquals` would help us to do the assertions without sleep.

Reply:
Ended up using assertEventuallyEquals, it works.